src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -195,8 +203,19 @@ def type_text(self):
         # Set reader to free format.
         string_reader.set_format(FortranFormat(True, False))
         ParserFactory().create(std="f2008")
-        ptree = Fortran2003.Declaration_Construct(
-            string_reader)
+        try:
+            ptree = Fortran2003.Declaration_Construct(
+                string_reader)
+        except Exception as err:
+            raise NotImplementedError(
+                f"Cannot extract the declaration part from UnknownFortranType "
+                f"'{self._declaration}' because parsing failed.") from err
+        if not ptree or not isinstance(ptree,
+                                       (Fortran2003.Declaration_Construct,
+                                        Fortran2003.Type_Declaration_Stmt)):
+            raise NotImplementedError(
+                f"Cannot extract the declaration part from UnknownFortranType "
+                f"'{self._declaration}'")
```

Comment:
This method appears only to be used when comparing two `UnknownFortranTypes` for equality. Perhaps we never had a test case that did this for the common types. I used `NotImplementedError` because it really is just not implemented. Looking at #2137, I see that that proposes removing the name from the declaration text which makes sense but may be problematic when the 'declaration' isn't a normal declaration.
I'll try extending this method so that it does cope with common blocks and save statements and see how that goes...

Reply:
That was quite easy so we now do support common and save statements and checking their types for equality.